Add garbage collector - Prep code

mokagio opened the pull request:
This PR adds all the noisy infrastructure code on which the garbage collector script will rely, so that it'll be easier to review its code. It's purpose is to make #6 easier to review.

- Bundler setup
- RSpec setup
- JSON fixtures

mokagio committed:
Setup rspec and webmock via Bundler

We'll use these to write the unit tests for the garbage collector.

mokagio committed:
Add fixture methods for the JSON responses of the APIs we'll touch

Doing this in a dedicated commit so that the diff of the commits
actually adding the code will be easier to read

AliSoftware commented on Gemfile:
```diff
@@ -0,0 +1,12 @@
+
+source "https://rubygems.org"
+
+ruby File.read('.ruby-version').strip
```
smart!

AliSoftware commented on spec/github_client_spec.rb:
```diff
@@ -0,0 +1,57 @@
+def get_pr_response(state)
+  # This is only a fragment of the full API response with the data relevant the
+  # job we have to do.
```
```suggestion
  # This is only a fragment of the full API response with the data relevant for
  # the job we have to do.
```

AliSoftware commented on spec/spec_helper.rb:
```diff
@@ -0,0 +1,102 @@
+  # is tagged with `:focus`, all examples get run. RSpec also provides
+  # aliases for `it`, `describe`, and `context` that include `:focus`
+  # metadata: `fit`, `fdescribe` and `fcontext`, respectively.
+  config.filter_run_when_matching :focus
```
TIL! worth enabling imho

AliSoftware commented on spec/spec_helper.rb:
```diff
@@ -0,0 +1,102 @@
+
+  # This setting enables warnings. It's recommended, but in some cases may
+  # be too noisy due to issues in dependencies.
+  config.warnings = true
```
worth enabling imho

loremattei replied:
I agree. Especially at this early stage.

AliSoftware reviewed (approved):
LGTM, although I'd probably enable some of the optional/commented config options for RSpec here for good measure (better use the good practices from the start!)

loremattei reviewed (approved):
Other than Olivier's suggestion, this looks good to me.

mokagio committed:
Enable most RSpec recommendations for a better and safer dev experience

Kept the slowest examples log disabled because this project is small in
size and if don't foresee its test suite becoming so long that we'll
need to tune its execution time.

mokagio commented:
Thanks folks for the reviews.

I'll admit that, in the midsts of trying to get the tests in place, I glossed over the RSpec recommendations. Definitely work enabling, so thank you very much for pointing them out 🙇
